### PR TITLE
Add GetFrameBeforeSample and GetFrameAfterSample to AnalyzerResults

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: include/AnalyzerResults.h [hunk 1]
```diff
@@ -32,79 +32,92 @@ class LOGICAPI Frame
...
+  public: // don't use
+    bool DoBubblesAppearOnChannel( Channel& channel );
+    bool DoMarkersAppearOnChannel( Channel& channel );
+    bool GetFramesInRange( S64 starting_sample_inclusive, S64 ending_sample_inclusive, U64* first_frame_index, U64* last_frame_index );
+    bool GetMarkersInRange( Channel& channel, S64 starting_sample_inclusive, S64 ending_sample_inclusive, U64* first_marker_index,
+                            U64* last_marker_index );
+    void GetMarker( Channel& channel, U64 marker_index, MarkerType* marker_type, U64* marker_sample );
+    U64 GetNumMarkers( Channel& channel );
+    S64 GetFrameBeforeSample( S64 sample_number );
+    S64 GetFrameAfterSample( S64 sample_number );
```
Comment (huffman) [inline include/AnalyzerResults.h]: I wasn't sure if we used optional in our public API, so I didn't use that here. We could, though.

Comment (Marcus10110) [reply include/AnalyzerResults.h]: let's make it an out variable like you mentioned, with a bool return value.

**Comment by michaelmaniscalco** [comment]
Hi Mark.  For whatever reason I'm suddenly getting these emails again.  My email must be on some list still.- MichaelSent from my T-Mobile 4G LTE Device
-------- Original message --------From: Mark <notifications@github.com> Date: 11/8/19  17:18  (GMT-05:00) To: saleae/AnalyzerSDK <AnalyzerSDK@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [saleae/AnalyzerSDK] Add GetFrameBeforeSample and GetFrameAfterSample to AnalyzerResults (#19) @Marcus10110 approved this pull request.

lgtm!

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/saleae/AnalyzerSDK/pull/19?email_source=notifications\u0026email_token=ACSE2Q57YGO2FBSBCBN65YDQSXQR3A5CNFSM4JK3B222YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCK7LZSA#pullrequestreview-314490056",
"url": "https://github.com/saleae/AnalyzerSDK/pull/19?email_source=notifications\u0026email_token=ACSE2Q57YGO2FBSBCBN65YDQSXQR3A5CNFSM4JK3B222YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCK7LZSA#pullrequestreview-314490056",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by huffman** [comment]
@michaelmaniscalco You are probably receiving the emails because you are on the watch list. You should be able to "unwatch" the repository by going to https://github.com/saleae/AnalyzerSDK and clicking the "Unwatch" button at the top.

**Pull request merged**